### PR TITLE
chore: use templated noxfile.py

### LINKED ISSUE #324: Noxfile is not updated with Sphinx version pin
`noxfile.py` is not updated with Sphinx version pin, please help take a look!

### PR DESCRIPTION
**Pull request by parthea**
Please review PR #326 before this one.

Fixes #324 🦕

### CODE REVIEW

File: owlbot.py [hunk 1]
```diff
@@ -89,15 +89,72 @@ def get_staging_dirs(
...
+# Fix noxfile.py
+# ----------------------------------------------------------------------------
+
+def place_before(path, text, *before_text, escape=None):
+    replacement = "\n".join(before_text) + "\n" + text
+    if escape:
+        for c in escape:
+            text = text.replace(c, '\\' + c)
+    s.replace([path], text, replacement)
+
```
[the remark this answers is not in the document]
Comment (parthea) [reply owlbot.py:104]: I think so. I actually saw this in [python-bigquery-sqlalchemy](https://github.com/googleapis/python-bigquery-sqlalchemy ). The original author is @jimfulton. I've pinged him here:
https://github.com/googleapis/python-bigquery-sqlalchemy/commit/8d9ee7c24b00fc9f15acd78e0ea9f8a1f49538c4#r52365572

[the remark this answers is not in the document]
Comment (parthea) [reply owlbot.py:104]: Filed https://github.com/googleapis/synthtool/issues/1125

[the remark this answers is not in the document]
Comment (parthea) [reply owlbot.py:104]: I'm going to resolve this comment as the feature is tracked in googleapis/synthtool#1125.

File: owlbot.py [hunk 2]
```diff
@@ -89,15 +89,72 @@ def get_staging_dirs(
     cov_level=99
 )
 
-s.move(templated_files, excludes=[".coveragerc", "noxfile.py"])
+s.move(templated_files, excludes=[".coveragerc"])
+
+# ----------------------------------------------------------------------------
+# Fix noxfile.py
+# ----------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (parthea) [reply owlbot.py]: Done.

**Comment by parthea** [comment]
This is ready for review

**Review by tseaver — APPROVED** [review]

**Comment by tseaver** [comment]
@parthea, @kolea2 the failing samples look flaky to me:  adding a cluster to a new instance is inherently tricksy, and needs more retry magyk fairy dust sprinkled on.

**Comment by tseaver** [comment]
Superseded by #344.

**Pull request closed without merging**